### DISCUSSION TITLE
Server Action Error: Unexpected identifier ("action-browser")

### DISCUSSION THREAD

**Discussion by john-prutton** [vercel/next.js#58245] [Help]
### Summary

When I make a db call from within a server action I get the following error:
![image](https://github.com/vercel/next.js/assets/19376890/2d46475e-71b3-40e7-a274-d0eb5214d9f8)

Interestingly enough, I get no error when making the same call from within a RSC -- without the 'use server' directive -- or from within an api route.

It may be important to note im working in a turborepo with pnpm workspaces. The drizzle db object is coming from another package via 
```ts
import {db} from "database"
```

### Additional information

```
Next.js: 14.0.1

@libsql/client: 0.3.6

drizzle-orm: 0.28.6
```

**Comment by msalahz** [comment]
any updates here

**Comment by msalahz** [comment]
I'm getting the same error when I try to user firebase to upload file in a server action

**Reply by MaxenceAdnot** [reply]
I have the same issue with firebase-admin and I managed to get it working with adding the following to next.config.js 

```
    experimental: {
      esmExternals: false,
    },
```

One other solution I found that does not involve the addition of a experimental flag is to pass the server action function through a prop from the parent Server Component down to the Client Component.

**Comment by nomnivore** [comment]
I'm getting the same error - following the official dashboard app workshop, but using drizzle/turso instead of `@vercel/postgres`. The `createInvoice` server action worked perfectly, but I'm running into this issue with the `updateInvoice` server action.
